tests/libs/estdlib/test_tcp_socket.erl

```diff
@@ -0,0 +1,187 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Fred Dushin <fred@dushin.net>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_tcp_socket).
+
+-export([test/0]).
+
+test() ->
+    ok = test_echo_server(),
+    % ok = test_handoff_receive(),
+    case get_otp_version() of
+        atomvm ->
+            ok = test_abandon_select();
+        _ ->
+            ok
+    end,
+    ok.
+
+test_echo_server() ->
+    Port = 44404,
+    {ok, ListenSocket} = socket:open(inet, stream, tcp),
+
+    ok = socket:setopt(ListenSocket, {socket, reuseaddr}, true),
+    ok = socket:setopt(ListenSocket, {socket, linger}, #{onoff => true, linger => 0}),
+
+    ok = socket:bind(ListenSocket, #{
+        family => inet, addr => loopback, port => Port
+    }),
+
+    ok = socket:listen(ListenSocket),
+
+    Self = self(),
+    spawn(fun() ->
+        Self ! ready,
+        accept(Self, ListenSocket)
+    end),
+
+    %% give accept a chance to run
+    timer:sleep(100),
```

Comment:
This is asking for trouble with the CI. I'll loop until connection succeeds, with a timeout instead.

Reply:
Added a loop.